scripts/updateReleaseNotes/changelogsAndTags.js

```diff
@@ -0,0 +1,115 @@
+// @ts-check
+/// <reference path="./index.d.ts" />
+
+const path = require('path');
+const fs = require('fs');
+const execSync = require('child_process').execSync;
+
+const MILLIS_PER_DAY = 1000 * 60 * 60 * 24;
+
+/**
+ * Builds a map of changelog tags to entries defined in all CHANGELOG.json files.
+ * @param {number} [maxAgeDays] - If provided, only include entries less than this many days old.
+ * Otherwise get all entries.
+ * @returns {Map<string, IChangelogEntry>}
+ */
+function getTagToChangelogMap(maxAgeDays) {
+  console.log(`Getting changelog entries${maxAgeDays ? ` up to ${maxAgeDays} days old` : ''}...`);
+
+  /** @type {Map<string, IChangelogEntry>} */
+  const map = new Map();
+
+  forEachFileRecursive(undefined, 'CHANGELOG.json', result => {
+    /** @type {IChangelog} */
+    const changelog = JSON.parse(result.content);
+    for (const entry of changelog.entries) {
+      if (isNewEnough(entry.date, maxAgeDays)) {
+        entry.name = changelog.name;
+        map.set(entry.tag, entry);
+      } else {
+        // changelog entries should be in reverse chronological order, so stop after the first one
+        // that's too old
+        break;
+      }
+    }
+  });
+
+  console.log(`Found changelog entries for ${map.size} tags.\n`);
+
+  return map;
+}
+
+/**
+ * Gets all the tags in a repo using 'git tag'.
+ * @param {number} [maxAgeDays] - If provided, only include entries less than this many days old.
+ * Otherwise get all entries.
+ * @returns {string[]} List of tags
+ */
+function getTags(maxAgeDays) {
+  console.log(`Getting tags${maxAgeDays ? ` up to ${maxAgeDays} days old` : ''}...`);
+
+  const cmd = 'git for-each-ref --sort=-creatordate --format="%(refname:short) -- %(creatordate)" refs/tags';
```

Comment:
Double quotes inside this are the bug fix for Windows. Also I was previously using `committerdate` which isn't set for all tag types, but `creatordate` is.